src/components/IconInput/IconInput.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from 'react-emotion';
+import { size } from 'polished';
+import Input from '../Input';
+
+import { themePropType } from '../../util/shared-prop-types';
+
+const LEFT = 'left';
+const RIGHT = 'right';
+
+const iconBaseStyles = ({ theme }) => css`
+  label: icon-input__icon;
+  position: absolute;
+  ${size(theme.spacings.mega)};
+  top: 50%;
+  transform: translateY(-50%);
+`;
+
+const iconLeftStyles = ({ theme, iconPosition }) =>
+  iconPosition === LEFT &&
+  css`
+    label: icon-input__icon--left;
+    left: ${theme.spacings.kilo};
+    right: auto;
+  `;
+
+const iconRightStyles = ({ theme, iconPosition }) =>
+  iconPosition === RIGHT &&
+  css`
+    label: icon-input__icon--right;
+    right: ${theme.spacings.kilo};
+    left: auto;
+  `;
+
+const inputLeftStyles = ({ theme, iconPosition }) =>
+  iconPosition === LEFT &&
+  css`
+    label: icon-input__input--left;
+    padding-left: calc(
+      ${theme.spacings.kilo} + ${theme.spacings.mega} + ${theme.spacings.kilo}
+    );
+  `;
+
+const inputRightStyles = ({ theme, iconPosition }) =>
+  iconPosition === RIGHT &&
+  css`
+    label: icon-input__input--right;
+    padding-right: calc(
+      ${theme.spacings.kilo} + ${theme.spacings.mega} + ${theme.spacings.kilo}
+    );
+  `;
+
+/**
+ * Renders inputs or selects that have an icon overlay. Takes the icon as a
```

Comment:
I guess this now only handles `input`s and not `select`s?

Reply:
Yes, I should update the description.